### ISSUE TITLE
New-HPOVServerProfile -ApplianceConnection issue

### ISSUE THREAD

**Issue by Chris-eanes** [HewlettPackard/POSH-HPOneView#400]
Please fill in as much information as possible to help resolve your issue.

### Expected Behavior

Create a new server profile on a non default OV appliance connection:

```PowerShell
New-HPOVServerProfile -Name $SPName -ServerProfileTemplate $SPT -ApplianceConnection $APPCONN -AssignmentType Server -Server $Server
```

### Actual Behavior

It appears the attempt to create the SP happens on the default $ConnectedSessions instead of the specified appliance connection.  The default connection always works (no matter which OV is the default)

### Steps to reproduce

```PowerShell
$APPCONN = DCS02.zbook.net

The "New-HPOVServerProfile" does not use the -ApplianceConnection parameter of $APPCONN and uses the default appliance:


VERBOSE: [SEND-HPOVREQUEST] Processing 'DCS01.zbook.net' appliance connection request. 1 of 1
VERBOSE: [SEND-HPOVREQUEST] Requested URI '/rest/enclosure-groups/4cb54400-b7d7-4ca0-896b-591107bed6dd' to 'DCS01.zbook.net'
VERBOSE: [RESTCLIENT] Building new [System.Net.HttpWebRequest] object for GET https://DCS01.zbook.net/rest/enclosure-groups/4cb54400-b7d7-4ca0-896b-591107bed
6dd
VERBOSE: [SEND-HPOVREQUEST] Request: GET https://DCS01.zbook.net/rest/enclosure-groups/4cb54400-b7d7-4ca0-896b-591107bed6dd

VERBOSE: [SEND-HPOVREQUEST] Exception Message: The remote server returned an error: (404) Not Found.
{"errorCode":"RESOURCE_NOT_FOUND","message":"Resource not found, ID/URI = 4cb54400-b7d7-4ca0-896b-591107bed6dd.",
"details":"The requested resource not found, ID = 4cb54400-b7d7-4ca0-896b-591107bed6dd.","recommendedActions":["Confirm resource ID/URI and try the call agai
n."],"errorSource":null,"nestedErrors":[],"data":{}}
VERBOSE: [SEND-HPOVREQUEST] Response Status: HTTP 404 [Not Found]
```

### Version Information

```PowerShell
DCS01.zbook.net                     DCS02.zbook.net                     LibraryVersion Path                                                                 
---------------                     ---------------                     -------------- ----                                                                 
ApplianceVersion: 4.20.01.380241.00 ApplianceVersion: 4.20.01.380241.00 4.20.1971.1960 C:\Program Files\WindowsPowerShell\Modules\HPOneView.420\4.20.1971...



Name                           Value                                                                                                                        
----                           -----                                                                                                                        
PSVersion                      5.1.17763.316                                                                                                                
PSEdition                      Desktop                                                                                                                      
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                      
BuildVersion                   10.0.17763.316                                                                                                               
CLRVersion                     4.0.30319.42000                                                                                                              
WSManStackVersion              3.0                                                                                                                          
PSRemotingProtocolVersion      2.3                                                                                                                          
SerializationVersion           1.1.0.1                                                                                                                      
```

**Comment by ChrisLynchHPE** [comment]
Please use `Get-HPOVCommandTrace` to collect a verbose output capture, and supply the created file.

**Comment by Chris-eanes** [comment]
[1905102059_HPOV_transcript.log](https://github.com/HewlettPackard/POSH-HPOneView/files/3168519/1905102059_HPOV_transcript.log)

**Comment by Chris-eanes** [comment]
Line 1236 in the trace references the default appliance connection.

**Comment by Chris-eanes** [comment]
$conn1=Connect-HPOVMgmt -Hostname DCS01.zbook.net -Credential $cred
$conn2=Connect-HPOVMgmt -Hostname DCS02.zbook.net -Credential $cred

$SPT = Get-HPOVServerProfileTemplate -ApplianceConnection $conn2
$Server = $SPT | Get-HPOVServer -NoProfile -ApplianceConnection $conn2 | Select -First 1


Get-HPOVCommandTrace -ScriptBlock {


$SPT = Get-HPOVServerProfileTemplate -ApplianceConnection $conn2
$Server = $SPT | Get-HPOVServer -NoProfile -ApplianceConnection $conn2 | Select -First 1

New-HPOVServerProfile -Name 'TestSP' -ServerProfileTemplate $SPT -ApplianceConnection $conn2 -AssignmentType Server -Server $Server -Async

}

**Comment by ChrisLynchHPE** [comment]
Thanks.  I found the issue and will have this fixed in the next update.

I also found this issue in the 4.10 library.

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 4.20.2068.2204](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v4.20.2068.2204), [PowerShell Gallery HPOneView.420](https://www.powershellgallery.com/packages/HPOneView.420/4.20.2068.2204), and the [Release 4.10.2059.3121](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v4.10.2059.3121), [PowerShell Gallery HPOneView.410](https://www.powershellgallery.com/packages/HPOneView.410/4.10.2059.3121)

Please verify before this issue is closed.

**Comment by Chris-eanes** [comment]
Issue is resolved
